msticpy/data/drivers/local_velociraptor_driver.py

```
@@ -0,0 +1,222 @@
+            A DataFrame (if successful) or
+            the underlying provider result if an error.
+
+        """
+        del kwargs
+        if not self.data_files:
+            self.connect()
+
+        if query not in self.data_files:
+            raise MsticpyDataQueryError(
+                f"No data loaded for query {query}.",
+                "Please check that the data loaded from the log paths",
+                "has this data type.",
+            )
+        return self._cached_query(query)
+
+    @lru_cache(maxsize=256)
+    def _cached_query(self, query: str) -> pd.DataFrame:
+        iter_data_files = (
+            tqdm(self.data_files[query]) if self._progress else self.data_files[query]
+        )
+        dfs = [pd.read_json(file, lines=True) for file in iter_data_files]
+        query_df = pd.concat(dfs)
+
+        logger.info("Query %s, returned %d rows", query, len(query_df))
+        return query_df
+
+    def query_with_results(self, query, **kwargs):
+        """Return query with fake results."""
+        return self.query(query, **kwargs), "OK"
+
+    @property
+    def driver_queries(self) -> List[Dict[str, Any]]:
+        """
+        Return dynamic queries available on connection to data.
+
+        Returns
+        -------
+        List[Dict[str, Any]]
+            List of queries with properties: "name", "query", "container"
+            and (optionally) "description"
+
+        Raises
+        ------
+        MsticpyNotConnectedError
+            If called before driver is connected.
+
+        """
+        if not self.connected:
+            self.connect()
+        if self.data_files:
+            return [
+                {
+                    "name": query,
+                    "query": query,
+                    "query_paths": "velociraptor",
+                    "description": f"Velociraptor {query} table.",
+                }
+                for query in self.data_files
+            ]
+        return []
```

Since `self.connect` should be assigning `self.data_files`, if `self.data_files` is none then it should throw an exception (or at least generate an error message, rather than fail silently).